src/lib/mina_base/snapp_generators.ml

```diff
@@ -266,11 +266,11 @@ let gen_balance_change ?balances_tbl ?permissions_auth (account : Account.t) =
       (* if positive, the account balance does not impose a constraint on the magnitude; but
          to avoid overflow over several Party.t, we'll limit the value
       *)
-      let%map magnitude =
+      let%map (magnitude : Currency.Amount.t) =
         Currency.Amount.gen_incl Currency.Amount.zero
           (Currency.Amount.of_int 100_000_000_000)
       in
-      Currency.Signed_poly.{ magnitude; sgn = Sgn.Pos }
+      ({ magnitude; sgn = Sgn.Pos } : Currency.Amount.Signed.t)
```

Comment:
I see why this style is clearer as to what types you're talking about (it feels better honestly), but it's different from the rest of our code. So I'm mixed on if we should allow this through. cc @mrmr1993 for thoughts

Reply:
This was done because adding the `fields` ppx to `Currency.Signed_poly.t` caused compile errors in `snapp_generators` on a call to `Fn.id()`. The actual compile error was something along the lines of:
```ocaml
ile "src/lib/mina_base/snapp_generators.ml", line 633, characters 24-29:
633 |       ~f_balance_change:Fn.id ()
                              ^^^^^
Error: This expression has type
         (('a, 'b) Currency.Signed_poly.t -> 'a, Sgn.t)
         Currency.Signed_poly.t ->
         (('a, 'b) Currency.Signed_poly.t -> 'a, Sgn.t)
         Currency.Signed_poly.t
       but an expression was expected of type
         (('a, 'b) Currency.Signed_poly.t -> 'a, Sgn.t)
         Currency.Signed_poly.t -> Currency.Amount.Signed.t
       Type
         (('a, 'b) Currency.Signed_poly.t -> 'a, Sgn.t)
         Currency.Signed_poly.t
       is not compatible with type
         Currency.Amount.Signed.t =
           (Currency.Amount.t, Sgn.t) Currency.Signed_poly.t 
       Type ('a, 'b) Currency.Signed_poly.t -> 'a is not compatible with type
         Currency.Amount.t 
```
This was done to fix that compile error.

Reply:
ah interesting, 💯